components/schema-migrator/migrations/director/20220203161810_ord_internal_visibility.up.sql

```diff
@@ -0,0 +1,214 @@
+BEGIN;
+
+ALTER TABLE bundle_references
+ADD COLUMN visibility varchar;
+
+UPDATE bundle_references br
+SET visibility = (SELECT visibility FROM api_definitions WHERE api_def_id = api_definitions.id)
+WHERE br.api_def_id IS NOT NULL;
+
+UPDATE bundle_references br
+SET visibility = (SELECT visibility FROM event_api_definitions WHERE event_def_id = event_api_definitions.id)
+WHERE br.event_def_id IS NOT NULL;
+
+-- all null records from APIs/Events mean that those APIs/Events have come from graphql flow - meaning that they are public, because the concept of private visibility is only ORD scoped
+UPDATE bundle_references
+SET visibility = 'public'
+WHERE visibility IS NULL;
+
+ALTER TABLE bundle_references
+ALTER COLUMN visibility SET NOT NULL;
+
+
+--- changes to tenants_apis and tenants_events ORD views
+
+DROP VIEW IF EXISTS tenants_specifications; -- drop tenant_specifications because tenant_apis and tenant_events depends on it. There are no changes on this view.
+DROP VIEW IF EXISTS tenants_apis;
+DROP VIEW IF EXISTS tenants_events;
+
+-- cast visibility field to text
```

Comment:
Due to the introduction of the new claim and the corresponding `@EdmProtectedBy` annotation in the ORD Service. 

When the `WHERE` clause based on the claims is constructed and the `visibility` field is added there, it expects a string (because the `visibility` field is of type string in the Java code) but the matching value from the view is the enum type and error is thrown due to mismatching types.

Part of the error: `No operator matches the given name and argument types. You might need to add explicit type casts.`